AsyncDisplayKit/Layout/ASDimension.mm

```diff
@@ -11,66 +11,243 @@
+      return ASDimensionMake(ASDimensionTypeFraction, dimension.floatValue);
+    }
+  }
+
+  // Assert as parsing went wrong
+  ASDisplayNodeCAssert(NO, @"Parsing dimension failed");
+  return ASDimensionAuto;
+}
+
+ASDimension ASDimensionMakeWithPoints(CGFloat points)
 {
   ASDisplayNodeCAssertPositiveReal(@"Points", points);
-  return ASRelativeDimensionMake(ASRelativeDimensionTypePoints, points);
+  return ASDimensionMake(ASDimensionTypePoints, points);
 }
 
-ASRelativeDimension ASRelativeDimensionMakeWithFraction(CGFloat fraction)
+ASDimension ASDimensionMakeWithFraction(CGFloat fraction)
 {
-  // ASDisplayNodeCAssert( 0 <= fraction && fraction <= 1.0, @"ASRelativeDimension fraction value (%f) must be between 0 and 1.", fraction);
-  return ASRelativeDimensionMake(ASRelativeDimensionTypeFraction, fraction);
+  // ASDisplayNodeCAssert( 0 <= fraction && fraction <= 1.0, @"ASDimension fraction value (%f) must be between 0 and 1.", fraction);
+  return ASDimensionMake(ASDimensionTypeFraction, fraction);
 }
 
-ASRelativeDimension ASRelativeDimensionCopy(ASRelativeDimension aDimension)
+ASDimension ASDimensionCopy(ASDimension aDimension)
 {
-  return ASRelativeDimensionMake(aDimension.type, aDimension.value);
+  return ASDimensionMake(aDimension.type, aDimension.value);
 }
 
-BOOL ASRelativeDimensionEqualToRelativeDimension(ASRelativeDimension lhs, ASRelativeDimension rhs)
+BOOL ASDimensionEqualToDimension(ASDimension lhs, ASDimension rhs)
 {
   return lhs.type == rhs.type && lhs.value == rhs.value;
 }
 
-NSString *NSStringFromASRelativeDimension(ASRelativeDimension dimension)
+NSString *NSStringFromASDimension(ASDimension dimension)
 {
   switch (dimension.type) {
-    case ASRelativeDimensionTypePoints:
+    case ASDimensionTypePoints:
       return [NSString stringWithFormat:@"%.0fpt", dimension.value];
-    case ASRelativeDimensionTypeFraction:
+    case ASDimensionTypeFraction:
       return [NSString stringWithFormat:@"%.0f%%", dimension.value * 100.0];
+    case ASDimensionTypeAuto:
+      return @"Auto";
   }
 }
 
-CGFloat ASRelativeDimensionResolve(ASRelativeDimension dimension, CGFloat parent)
+CGFloat ASDimensionResolve(ASDimension dimension, CGFloat autoSize, CGFloat parent)
 {
   switch (dimension.type) {
-    case ASRelativeDimensionTypePoints:
+    case ASDimensionTypeAuto:
+      return autoSize;
+    case ASDimensionTypePoints:
       return dimension.value;
-    case ASRelativeDimensionTypeFraction:
+    case ASDimensionTypeFraction:
       return dimension.value * parent;
   }
 }
 
+@implementation NSNumber (ASDimension)
+
+- (ASDimension)as_points
+{
+  return ASDimensionMake(ASDimensionTypePoints, self.floatValue);
+}
+
+- (ASDimension)as_fraction
+{
+  return ASDimensionMake(ASDimensionTypeFraction, self.floatValue);
+}
+
+@end
+
+#pragma mark - ASRelativeSize
+
+ASRelativeSize ASRelativeSizeMake(ASDimension width, ASDimension height)
+{
+  ASRelativeSize size; size.width = width; size.height = height; return size;
```

Comment:
Nit: this is actually less readable all on one line. please break it up over a few lines